dlt-database/migrations/0001-init_db.ts

```diff
@@ -0,0 +1,121 @@
+  // write upgrade logic as parameter of queryFn
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`users\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`gradido_id\` char(36) DEFAULT NULL,
+      \`pubkey\` binary(32) NOT NULL,
+      \`created_at\` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+      \`confirmed_at\` datetime DEFAULT NULL,
+      PRIMARY KEY (\`id\`),
+      UNIQUE KEY \`gradido_id\` (\`gradido_id\`),
+      UNIQUE KEY \`pubkey\` (\`pubkey\`)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;`)
+
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`accounts\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`user_id\` int(10) unsigned DEFAULT NULL,
+      \`derivation_index\` int(10) unsigned NOT NULL,
+      \`pubkey\` binary(32) NOT NULL,
+      \`type\` tinyint unsigned NOT NULL,
+      \`created_at\` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+      \`confirmed_at\` datetime DEFAULT NULL,
+      \`balance\` decimal(40,20) NOT NULL DEFAULT 0,
+      \`balance_date\` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP, 
+      PRIMARY KEY (\`id\`),
+      UNIQUE KEY \`pubkey\` (\`pubkey\`),
+      FOREIGN KEY (\`user_id\`) REFERENCES users(id)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+  `)
+
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`communities\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`iota_topic\` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
+      \`pubkey\` binary(32) NOT NULL,
+      \`privkey\` binary(32) DEFAULT NULL,
+      \`chaincode\` binary(32) DEFAULT NULL,
+      \`foreign\` tinyint(4) NOT NULL DEFAULT true,
+      \`gmw_account_id\` int(10) unsigned DEFAULT NULL,
+      \`auf_account_id\` int(10) unsigned DEFAULT NULL,
+      \`created_at\` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+      \`confirmed_at\` datetime DEFAULT NULL,
+      PRIMARY KEY (\`id\`),
+      UNIQUE KEY \`pubkey\` (\`pubkey\`),
+      FOREIGN KEY (\`gmw_account_id\`) REFERENCES accounts(id),
+      FOREIGN KEY (\`auf_account_id\`) REFERENCES accounts(id)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;`)
+
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`accounts_communities\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`account_id\` int(10) unsigned NOT NULL,
+      \`community_id\` int(10) unsigned NOT NULL,
+      \`valid_from\` datetime NOT NULL,
+      \`valid_to\` datetime DEFAULT NULL,
+      PRIMARY KEY (\`id\`),
+      FOREIGN KEY (\`account_id\`) REFERENCES accounts(id),
+      FOREIGN KEY (\`community_id\`) REFERENCES communities(id)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;`)
+
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`transaction_recipes\` (
+      \`id\` bigint unsigned NOT NULL AUTO_INCREMENT,
+      \`iota_message_id\` binary(32) DEFAULT NULL,
+      \`signing_account_id\` int(10) unsigned NOT NULL,
```

Comment:
ok i think i have an idea, why you named it signing_account...

this is the sender_account and because the sender will sign the data in it you name it like this way to give on usage the information to treat with signed data, right?

But is this helpfull for a column-name of a foreignkey and loosing by the way the business information to whom the data belongs to?

Reply:
Do you now what's inside body_bytes? 
That is the transaction body serialized from protobuf. 
It contain all business data. The columns I have carefull choosen are for optimal processing from getting transaction from backend, to put together the Transaction Object for iota and validate transaction and then for received transaction validate again and let backend know the transaction is valid. 
I will need the signing account to able to check the signature. I'm not loosing any data.

Reply:
![Business Logic Overview](https://github.com/gradido/gradido/blob/iota5_dlt-database/docu/Concepts/DLT/overview.md)